ts/features/itwallet/common/components/ItwCredentialClaim.tsx

```diff
@@ -30,6 +31,69 @@ import { ItwCredentialStatus } from "../utils/itwTypesUtils";
 import { clipboardSetStringWithFeedback } from "../../../../utils/clipboard";
 import { HIDDEN_CLAIM_TEXT } from "../utils/constants.ts";
 import { CREDENTIALS_MAP, trackCopyListItem } from "../../analytics";
+import { ItwNestedClaimsListItem } from "./ItwNestedClaimsListItem.tsx";
+
+/**
+ * Type for the configuration of nested credential summaries.
+ */
+type NestedCredentialSummaryConfig = {
+  summaryLabelId: string;
+  summaryValueId: string;
+};
+
+/**
+ * Map of nested credential summary configurations.
+ * This is used to define how to summarize nested credentials in the UI.
+ * Each key is the type of the nested credential, and the value is an object
+ * containing the IDs of the label and value to be displayed in the summary.
+ * Map more types here as needed.
+ */
+const itwNestedCredentialSummaryMap: Record<
+  string,
+  NestedCredentialSummaryConfig
+> = {
+  education_degrees: {
+    summaryLabelId: "programme_type_name",
+    summaryValueId: "degree_course_name"
+  },
+  education_enrollments: {
+    summaryLabelId: "programme_type_name",
+    summaryValueId: "degree_course_name"
+  }
+};
+
+/**
+ * Given a set of claims for a single nested item and its parent's claim ID,
+ * this function looks up the correct configuration and returns the
+ * appropriate summary fields.
+ *
+ * @param claimId The ID of the parent claim (e.g., "education_degrees").
+ * @param singleItemClaims The array of claims for the one nested item.
+ * @returns An object with the summaryLabel and summaryValue.
+ */
+const getNestedItemSummary = (
+  claimId: string,
+  singleItemClaims: Array<ClaimDisplayFormat>
+): { summaryLabel?: string; summaryValue?: string } => {
+  const summaryConfig = itwNestedCredentialSummaryMap[claimId];
+
+  if (!summaryConfig) {
+    return {};
+  }
+
+  const claimsMap = new Map(
+    singleItemClaims.map(claim => [claim.id, claim.value])
+  );
+
+  const summaryLabel = claimsMap.get(summaryConfig.summaryLabelId) as
+    | string
```

Comment:
Since we are casting, I'd do it at the map level:
```ts
  const claimsMap = new Map(
    singleItemClaims.map(claim => [claim.id, claim.value as string])
  );
```
Even though this might be dangerous, we are implying nested claim values are always strings.